TFNetwork.py

```diff
@@ -1411,31 +1411,45 @@ def get_norm_factor(self):
     self._prepare()
     return self._norm_factor
 
-  def _normalized_loss_value_per_seq(self, value):
+  def _normalized_value_per_seq(self, value, per_pos=False):
     """
-    :param tf.Tensor|None loss:
+    :param tf.Tensor|None value: (batch*time,) or (time*batch,)
+    :param bool per_pos: if true return tf.Tensor(B,T) else tf.Tensor(B)
     :return: (batch,) or None if loss is None
     :rtype: tf.Tensor|None
     """
     if value is None:
       return None
-    return self.loss.reduce_to_batch(value, normalize=True)
 
-  def get_normalized_loss_value_per_seq(self):
+    if per_pos:
+      value = tf.reshape(value, tf.shape(self.loss.output.get_sequence_mask()))  # (batch,time) or (time,batch)
```

Comment:
This is a very inefficient way to get the shape. Why do you need to create a sequence mask for that? Instead, do sth like:
```suggestion
      value = tf.reshape(value, tf.shape(self.loss.output.placeholder)[:2])  # (batch,time) or (time,batch)
```

Reply:
we adapted this from:
https://github.com/rwth-i6/returnn/blob/b3117f972bf0c6c4c607c4165034f7f1e257e87b/TFNetworkLayer.py#L5422

we changed both lines now